File: csharp/src/IO.Swagger/Model/CheckTransactionSettings.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using System.ComponentModel.DataAnnotations;
using SwaggerDateConverter = IO.Swagger.Client.SwaggerDateConverter;

namespace IO.Swagger.Model
{
    [DataContract]
    public partial class CheckTransactionSettings :  IEquatable<CheckTransactionSettings>, IValidatableObject
    {
        public CheckTransactionSettings(FieldInfo reader = default(FieldInfo), FieldInfo contactName = default(FieldInfo), FieldInfo contactTitle = default(FieldInfo), FieldInfo emailAddress = default(FieldInfo), string phoneDefault = default(string), FieldInfo fax = default(FieldInfo), string billingAddressDefault = default(string), FieldInfo billingAddress2 = default(FieldInfo), string billingCityDefault = default(string), string billingStateDefault = default(string), string billingZipDefault = default(string), FieldInfo pONumber = default(FieldInfo), FieldInfo invoiceNumber = default(FieldInfo), FieldInfo businessName = default(FieldInfo), FieldInfo customerNumber = default(FieldInfo), FieldInfo referenceNote = default(FieldInfo), string defaultSECCode = default(string))
        {
            this.Reader = reader;
            this.ContactName = contactName;
            this.ContactTitle = contactTitle;
            this.EmailAddress = emailAddress;
            this.PhoneDefault = phoneDefault;
            this.Fax = fax;
            this.BillingAddressDefault = billingAddressDefault;
            this.BillingAddress2 = billingAddress2;
            this.BillingCityDefault = billingCityDefault;
            this.BillingStateDefault = billingStateDefault;
            this.BillingZipDefault = billingZipDefault;
            this.PONumber = pONumber;
            this.InvoiceNumber = invoiceNumber;
            this.BusinessName = businessName;
            this.CustomerNumber = customerNumber;
            this.ReferenceNote = referenceNote;
            this.DefaultSECCode = defaultSECCode;
        }

        [DataMember(Name="Reader", EmitDefaultValue=false)]
        public FieldInfo Reader { get; set; }

        [DataMember(Name="ContactName", EmitDefaultValue=false)]
        public FieldInfo ContactName { get; set; }

        [DataMember(Name="ContactTitle", EmitDefaultValue=false)]
        public FieldInfo ContactTitle { get; set; }

        [DataMember(Name="EmailAddress", EmitDefaultValue=false)]
        public FieldInfo EmailAddress { get; set; }

        [DataMember(Name="PhoneDefault", EmitDefaultValue=false)]
        public string PhoneDefault { get; set; }

        [DataMember(Name="Fax", EmitDefaultValue=false)]
        public FieldInfo Fax { get; set; }

        [DataMember(Name="BillingAddressDefault", EmitDefaultValue=false)]
        public string BillingAddressDefault { get; set; }

        [DataMember(Name="BillingAddress2", EmitDefaultValue=false)]
        public FieldInfo BillingAddress2 { get; set; }

        [DataMember(Name="BillingCityDefault", EmitDefaultValue=false)]
        public string BillingCityDefault { get; set; }

        [DataMember(Name="BillingStateDefault", EmitDefaultValue=false)]
        public string BillingStateDefault { get; set; }

        [DataMember(Name="BillingZipDefault", EmitDefaultValue=false)]
        public string BillingZipDefault { get; set; }

        [DataMember(Name="PONumber", EmitDefaultValue=false)]
        public FieldInfo PONumber { get; set; }

        [DataMember(Name="InvoiceNumber", EmitDefaultValue=false)]
        public FieldInfo InvoiceNumber { get; set; }

        [DataMember(Name="BusinessName", EmitDefaultValue=false)]
        public FieldInfo BusinessName { get; set; }

        [DataMember(Name="CustomerNumber", EmitDefaultValue=false)]
        public FieldInfo CustomerNumber { get; set; }

        [DataMember(Name="ReferenceNote", EmitDefaultValue=false)]
        public FieldInfo ReferenceNote { get; set; }

        [DataMember(Name="DefaultSECCode", EmitDefaultValue=false)]
        public string DefaultSECCode { get; set; }
    }
}

File: CodeSamples/nodejs/src/model/CheckTransactionSettings.js
import ApiClient from '../ApiClient';
import FieldInfo from './FieldInfo';

export default class CheckTransactionSettings {
    constructor() {
        this.Reader = new FieldInfo();
        this.ContactName = new FieldInfo();
        this.ContactTitle = new FieldInfo();
        this.EmailAddress = new FieldInfo();
        this.PhoneDefault = '';
        this.Fax = new FieldInfo();
        this.BillingAddressDefault = '';
        this.BillingAddress2 = new FieldInfo();
        this.BillingCityDefault = '';
        this.BillingStateDefault = '';
        this.BillingZipDefault = '';
        this.PONumber = new FieldInfo();
        this.InvoiceNumber = new FieldInfo();
        this.BusinessName = new FieldInfo();
        this.CustomerNumber = new FieldInfo();
        this.ReferenceNote = new FieldInfo();
        this.DefaultSECCode = '';
    }

    static constructFromObject(data, obj) {
        if (data) {
            obj = obj || new CheckTransactionSettings();

            if (Object.prototype.hasOwnProperty.call(data, 'Reader')) {
                obj['Reader'] = FieldInfo.constructFromObject(data['Reader']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'ContactName')) {
                obj['ContactName'] = FieldInfo.constructFromObject(data['ContactName']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'ContactTitle')) {
                obj['ContactTitle'] = FieldInfo.constructFromObject(data['ContactTitle']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'EmailAddress')) {
                obj['EmailAddress'] = FieldInfo.constructFromObject(data['EmailAddress']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'PhoneDefault')) {
                obj['PhoneDefault'] = ApiClient.convertToType(data['PhoneDefault'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'Fax')) {
                obj['Fax'] = FieldInfo.constructFromObject(data['Fax']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'BillingAddressDefault')) {
                obj['BillingAddressDefault'] = ApiClient.convertToType(data['BillingAddressDefault'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'BillingAddress2')) {
                obj['BillingAddress2'] = FieldInfo.constructFromObject(data['BillingAddress2']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'BillingCityDefault')) {
                obj['BillingCityDefault'] = ApiClient.convertToType(data['BillingCityDefault'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'BillingStateDefault')) {
                obj['BillingStateDefault'] = ApiClient.convertToType(data['BillingStateDefault'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'BillingZipDefault')) {
                obj['BillingZipDefault'] = ApiClient.convertToType(data['BillingZipDefault'], 'String');
            }
            if (Object.prototype.hasOwnProperty.call(data, 'PONumber')) {
                obj['PONumber'] = FieldInfo.constructFromObject(data['PONumber']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'InvoiceNumber')) {
                obj['InvoiceNumber'] = FieldInfo.constructFromObject(data['InvoiceNumber']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'BusinessName')) {
                obj['BusinessName'] = FieldInfo.constructFromObject(data['BusinessName']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'CustomerNumber')) {
                obj['CustomerNumber'] = FieldInfo.constructFromObject(data['CustomerNumber']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'ReferenceNote')) {
                obj['ReferenceNote'] = FieldInfo.constructFromObject(data['ReferenceNote']);
            }
            if (Object.prototype.hasOwnProperty.call(data, 'DefaultSECCode')) {
                obj['DefaultSECCode'] = ApiClient.convertToType(data['DefaultSECCode'], 'String');
            }
        }
        return obj;
    }
}
